src/configureAWS.ts

```diff
@@ -20,54 +20,48 @@ import * as aws from 'aws-sdk';
 import {logger} from './logger';
 import {AWSRegion} from './aws-regions';
 
-async function loadSharedIniFile(profile?: string) {
-  // note, profile might be undefined here and that's fine.
-  const credentials = new aws.SharedIniFileCredentials({profile});
-  //await credentials.refreshPromise();
-  aws.config.credentials = credentials;
+function getCredentialsProviderChain(profile?: string) {
+  const hasDotAWS = (awsUserDir && fs.existsSync(awsUserDir));
```

Comment:
Move into `if (profile)`

Reply:
I tried that and it just cluttered things and made it harder to keep `assumeRoleArn` compatible with `profile`.